bindings/jupyter/js/src/widgets/react-widget-builder.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import ReactDom from 'react-dom';
+import {DOMWidgetModel, DOMWidgetView} from '@jupyter-widgets/base';
+
+export default (Component, name) => {
+  const classes = {
+    [name + 'WidgetModel']: class extends DOMWidgetModel {},
+    [name + 'WidgetView']: class extends DOMWidgetView {
+      render = () => {
+        super.render(this);
+        this._update();
+        this.listenTo(this.model, 'change:props', this._update, this);
+      };
+
+      _update = () => {
+        const props = JSON.parse(this.model.get('props') || '{}');
```

Comment:
Add `try` `catch` for `JSON.parse` in case `this.model.get('props')` contains invalid JSON strings.

Reply:
This try catch should instead to be located inside the Python code,
```python
class CommonComponent(Widget):
    def __init__(self, props={}, **kwargs):
        super(CommonComponent, self).__init__()
        self.props = json.dumps(props)
```
because the only way there can be a malicious JSON is when the user inputed it inside Jupyter, and the Python wrapper will catch it first before the JS code does.